### PR TITLE
feat: add default implementation for providers

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

Fixes https://github.com/kubernetes-sigs/custom-metrics-apiserver/issues/150

### CODE REVIEW

**Comment by JorTurFer** [comment]
I think that some changes are required in the infra level because it has started the fails when I bumped deps and go version. 
Am I right @dgrisonnet ?

**Comment by dgrisonnet** [comment]
Yes, I need to update the infra image, I'll do that tomorrow

**Comment by JorTurFer** [comment]
> Yes, I need to update the infra image, I'll do that tomorrow

Sure, no rush at all, I asked just to know if there was something missing from my side :)

**Comment by dgrisonnet** [comment]
Do you need the dependency update in this PR? Otherwise I'd prefer having it in a separate PR if you don't mind since it is easier to track when I create the release note.

**Comment by JorTurFer** [comment]
> Do you need the dependency update in this PR? Otherwise I'd prefer having it in a separate PR if you don't mind since it is easier to track when I create the release note.

sore, omw

**Comment by JorTurFer** [comment]
I have renamed the metric names to follow tha k8s naming:
image
All the metrics go all together without underscores 😄

File: pkg/provider/defaults/default_metric_providers.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"sigs.k8s.io/custom-metrics-apiserver/pkg/provider"
+)
+
+type DefaultExternalMetricsProvider struct{}
+
+func (em DefaultExternalMetricsProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo {
+	return []provider.ExternalMetricInfo{
+		{
+			Metric: "externalmetric",
```
Comment (dgrisonnet) [inline pkg/provider/defaults/default_metric_providers.go:29]: resource names are always in plural form in k8s

File: pkg/provider/defaults/default_metric_providers.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		},
+	}
+}
+
+type DefaultCustomMetricsProvider struct{}
+
+func (cm DefaultCustomMetricsProvider) ListAllMetrics() []provider.CustomMetricInfo {
+	return []provider.CustomMetricInfo{
+		{
+			Metric: "custommetric",
```
Comment (dgrisonnet) [inline pkg/provider/defaults/default_metric_providers.go:39]: resource names are always in plural form in k8s

File: pkg/provider/defaults/default_metric_providers.go (hunk 1 above)
Comment (JorTurFer) [reply pkg/provider/defaults/default_metric_providers.go]: nice catch! I didn't notice it

**Pull request merged**